File: app/src/main/java/com/example/home/screen/SaveDataActivity.java
package com.example.home.screen;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.room.RoomDatabase;

import android.os.Bundle;
import android.widget.ArrayAdapter;

import com.example.home.adapter.SaveDataAdapter;
import com.example.home.databinding.ActivitySaveDataBinding;
import com.example.home.db.HitsItemDatabase;
import com.example.home.model.HitsItem;

import java.util.ArrayList;
import java.util.List;

public class SaveDataActivity extends AppCompatActivity {

    private ActivitySaveDataBinding binding;
    private SaveDataAdapter adapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySaveDataBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        List<HitsItem> items = HitsItemDatabase.getDatabaseInstance(this).getHitsItemDao().getData();
        adapter = new SaveDataAdapter();
        binding.rcvSaveData.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
        adapter.setItems(items);
        binding.rcvSaveData.setAdapter(adapter);

    }
}

File: app/src/main/java/com/example/assignment13_mvvm_kotlin/ui/activity/SaveDataActivity.kt
package com.example.assignment13_mvvm_kotlin.ui.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.activity.viewModels
import androidx.lifecycle.Observer
import com.example.assignment13_mvvm_kotlin.R
import com.example.assignment13_mvvm_kotlin.adapter.SaveDataAdapter
import com.example.assignment13_mvvm_kotlin.databinding.ActivitySaveDataBinding
import com.example.assignment13_mvvm_kotlin.viewmodel.MainViewModel
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class SaveDataActivity : AppCompatActivity() {

    private lateinit var binding : ActivitySaveDataBinding

    val viewModel : MainViewModel by viewModels()

    @Inject
    lateinit var adapter : SaveDataAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySaveDataBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.adapter = adapter
        setRecycler()
    }

    private fun setRecycler() {
        viewModel.getSavedHits()
        viewModel.savedList.observe(this, Observer {
            adapter.submitData(it)
        })
    }
}